### ISSUE TITLE
Async support for Interfaces is not implemented on master

### ISSUE THREAD

**Issue by Ekzer** [graphql-rust/juniper#605]
Hi there!
I am getting troubles to make this simple piece of code working : 
``` 

#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, juniper::GraphQLObject)]
pub struct Unit {
    id: String,
    duration: i32,
    productionNationalityCode: Vec<String>,
    format: Format
}

#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, juniper::GraphQLObject)]
pub struct Format {
    quality: Vec<String>,
    dst: bool,
    audio: Vec<String>,
    drmType: Vec<String>,
    distTechnology: Vec<String>,
    functionalType: Vec<String>,
    version: Vec<String>,
    comMode: Vec<String>,
    qad: bool
}

trait ContentInterface {
    fn id(&self) -> &str;

    fn as_unit(&self) -> Option<&Unit> { None }
}

impl ContentInterface for Unit {
    fn id(&self) -> &str {
        self.id.as_str()
    }

    fn as_unit(&self) -> Option<&Unit> { Some(self) }
}

juniper::graphql_interface!(<'a> &'a ContentInterface: () as "ContentInterface" where Scalar = <S> |&self| {
    field id() -> &str { self.id() }

    instance_resolvers: |_| {
        // The left hand side indicates the concrete type T, the right hand
        // side should be an expression returning Option<T>
        &Unit => self.as_unit()
    }
});

```

I a getting this error : 
``` 
40 | / juniper::graphql_interface!(<'a> &'a ContentInterface: () as "ContentInterface" where Scalar = <S> |&self| {
41 | |     field id() -> &str { self.id() }
42 | |
43 | |     instance_resolvers: |_| {
...  |
47 | |     }
48 | | });
   | |___^ cannot infer type for type parameter `S`
```

Thank you for your help in advance !
Best regards

**Comment by LegNeato** [comment]
Are you using master? Interfaces have yet to be ported to async master and thus don't currently work on master.

**Comment by Ekzer** [comment]
Hi @LegNeato ,
I am using the async one yes.
`juniper = {git = "https://github.com/graphql-rust/juniper", features=["async"]}`

So I have to go back on classic master if I understood well ?

**Comment by LegNeato** [comment]
Either you need to use the latest release on crates.io (which is sync but has interfaces and unions) or you need to wait until master has support for interfaces (happy to take a PR as well!).

Also, you should be able to delete `features=["async"]` with latest master.

**Comment by LegNeato** [comment]
Related to https://github.com/graphql-rust/juniper/issues/549

**Comment by Ekzer** [comment]
Thank you for your clear answer !
I will give a try then to give async support for interface, good challenge for me as I'm relatively new to this language :)

**Comment by tyranron** [comment]
Landed in #682
